File: AngularSocialNetwork.API/Models/Post.cs
namespace AngularSocialNetwork.API.Models
{
    public class Post
    {
        public int PostId { get; set; }
        public int UserId { get; set; }
        public string Text { get; set; }
        public int LikeCount { get; set; } = 0;
        public int RetweetCount { get; set; } = 0;
        public int CommentCount { get; set; } = 0;
        public DateTime CreatedDate { get; set; } = DateTime.Now;
    }
}

File: AngularSocialNetwork-SPA/src/app/_models/post.ts
export interface Post {
  id: number;
  postId: number;
  userId: number;
  username: string;
  firstName: string;
  lastName: string;
  text: string;
  date: Date;
  photo: string;
  likeCount: number;
  retweetCount: number;
  commentCount: number;
  isReposted: boolean;
  repostedFirstName: string;
  repostedLastName: string;
  liked: boolean;
  reposted: boolean;

  ownPost: boolean;
  
  comments: Post[];
}